### PR TITLE
9 delete comment

### PR DESCRIPTION
**Pull request by iwebsteraudio**

### CODE REVIEW

File: __tests__/int.test.js [hunk 1]
```diff
@@ -258,15 +258,37 @@ describe("PATCH - Increases votes by article_id", () => {
...
+    })
+  })
+  test("When given an invalid id, returns 400, Bad Request",()=>{
+    return request(app)
+    .delete("/api/comments/NaN")
+    .expect(400)
+    .then(({body})=>{
+      const { msg } = body;
+      expect(msg).toBe('Bad Request')
+    })
```
Comment (nc-richard-neat) [inline __tests__/int.test.js:292]: Nice testing :D

File: controllers/articles-controllers.js [hunk 2]
```diff
@@ -42,20 +43,27 @@ const postCommentsByArticleId = (req, res, next) => {
...
-  sendArticleById,
-  sendArticleData,
-  sendCommentsByArticleId,
-  postCommentsByArticleId,
-  patchArticleById,
+exports.deleteCommendById = (req, res, next) => {
+  const { comment_id } = req.params;
+  removeCommentById(comment_id)
+    .then((commentData) => {
+      if(commentData.length === 0){
```
Comment (nc-richard-neat) [inline controllers/articles-controllers.js:60]: Remember that any error handling logic should take place in the model 👍

File: models/articles-models.js [hunk 3]
```diff
@@ -58,4 +58,10 @@ const patchVotes = (article_id, inc_votes) => {
   })
 }
 
-module.exports = { fetchArticleById, fetchArticleData, fetchCommentData, postCommentData, patchVotes };
+exports.removeCommentById = (comment_id) =>{
+
+  return db.query(`DELETE FROM comments WHERE comment_id = $1 RETURNING *;`,[comment_id])
+  .then((result)=>{
+    return result.rows;
```
Comment (nc-richard-neat) [inline models/articles-models.js:65]: It's here we can check against the rows property for the error handling

File: controllers/articles-controllers.js [hunk 4]
```diff
@@ -42,20 +43,27 @@ const postCommentsByArticleId = (req, res, next) => {
...
+        return Promise.reject({
+          status: 404,
+          msg: 'comment not found'
+        })
+      }
+      res.status(204).send({ commentData });
```
Comment (nc-richard-neat) [inline controllers/articles-controllers.js:66]: We actually don't want to be sending anything back as a 204 is 'no content' on the response.

File: __tests__/int.test.js (hunk 1 above)
Comment (iwebsteraudio) [reply __tests__/int.test.js:292]: Just doing some housekeeping on nc news. Don't mind me if there are notifications.

**Pull request merged**